File: python/Garage.py
# nama
# luas
# daftarKendaraan
from Vehicle import Vehicle

class Garage():
    __nama = ""
    __luas = ""
    __daftarKendaraan = Vehicle()

    def __init__(self, nama="", luas="", daftarKendaraan=""):
        self.__nama = nama
        self.__luas = luas
        self.__daftarKendaraan = daftarKendaraan

    # getset
        
    def getnama(self):
        return self.__nama
    
    def getluas(self):
        return self.__luas
    
    def getdaftarKendaraan(self):
        return self.__daftarKendaraan

    
    def setnama(self, nama):
        self.__nama = nama

    def setluas(self, luas):
        self.__luas = luas

    def setdaftarKendaraan(self, daftarKendaraan):
        self.__daftarKendaraan = daftarKendaraan

    

File: c++/Garage.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Vehicle.cpp" // Assuming Vehicle class is defined in Vehicle.h

using namespace std;

class Garage : public Vehicle{
private:
    string nama;
    string luas;
    vector<Vehicle> daftarKendaraan;

public:
    Garage(string nama = "", string luas = "") {
        this->nama = nama;
        this->luas = luas;
    }

    string getNama() {
        return nama;
    }

    string getLuas() {
        return luas;
    }

    vector<Vehicle> getDaftarKendaraan() {
        return daftarKendaraan;
    }

    void setNama(string nama) {
        this->nama = nama;
    }

    void setLuas(string luas) {
        this->luas = luas;
    }

    void tambahKendaraan(const Vehicle& kendaraan) {
        daftarKendaraan.push_back(kendaraan);
    }
};
